add_libmagic.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+install_source() {
+    # install from source
+    # https://www.darwinsys.com/file/
+    # https://github.com/file/file/blob/FILE5_45/INSTALL#L51
+    (
+        version="file-5.45" &&
+        tmpfile="$(mktemp)" &&
+        curl -sSLo "${tmpfile}" "https://astron.com/pub/file/${version}.tar.gz" &&
+        tar xvf "${tmpfile}" &&
+        cd "${version}" &&
+        ./configure &&
+        make &&
+        make install &&
+        make installcheck &&
+        cd .. &&
+        rm -r "${version}"
+    ) || ( cd .. && false )
+}
+
+install_precompiled() {
+    # Mac https://formulae.brew.sh/formula/libmagic
+    # Debian https://packages.ubuntu.com/libmagic1
+    # Alpine https://pkgs.alpinelinux.org/package/libmagic
+    # RHEL https://git.almalinux.org/rpms/file
+    # Windows https://github.com/julian-r/file-windows
+    if [ -n "$(which brew)" ]; then
+        brew install libmagic
+    elif [ -n "$(which apt-get)" ]; then
+        apt-get update
+        apt-get install -y libmagic1
+    elif [ -n "$(which apk)" ]; then
+        apk add --update libmagic
+    elif [ -n "$(which dnf)" ]; then
+        dnf --setopt install_weak_deps=false -y install file-libs
+    else
+        # windows (no install, just download into current working directory)
+        # could also consider install using `pacman`: https://packages.msys2.org/base/mingw-w64-file
+        # which would require an update of copy_libmagic below to account for new magic.mgc paths
+        python <<EOF
+import platform, sysconfig, io, zipfile, urllib.request
+assert platform.system() == "Windows"
+machine = "x86" if sysconfig.get_platform() == "win32" else "x64"
+url = f"https://github.com/julian-r/file-windows/releases/download/v5.44/file_5.44-build104-vs2022-{machine}.zip"
```

Comment:
In case anyone else wondered, 5.44 is the latest release available there, as opposed to 5.45 for the source tarball in `install_source()`.